src/main/java/com/google/javascript/cl2dts/DeclarationGenerator.java

```diff
@@ -72,55 +80,120 @@
       return input.getString();
     }};
 
+  static class Options {
+    @Option(name = "-o", usage = "output to this file", metaVar = "OUTPUT")
+    String output = "-";
+
+    @Option(name = "--externs",
+        usage = "list of files to read externs definitions (as separate args)",
+        metaVar = "EXTERN...",
+        handler=StringArrayOptionHandler.class)
+    List<String> externs = new ArrayList<>();
+
+    @Option(name = "--skipParseExterns",
+        usage = "run faster by skipping the externs parsing (useful for tests)")
+    boolean skipParseExterns;
+
+    @Argument
+    List<String> arguments = new ArrayList<>();
+
+    private CompilerOptions getCompilerOptions() {
+      final CompilerOptions options = new CompilerOptions();
+      options.setClosurePass(true);
+      options.setCheckGlobalNamesLevel(CheckLevel.ERROR);
+      options.setCheckGlobalThisLevel(CheckLevel.ERROR);
+      options.setCheckTypes(true);
+      options.setInferTypes(true);
+      options.setIdeMode(true); // So that we can query types after compilation.
+      options.setErrorHandler(new ErrorHandler() {
+        @Override
+        public void report(CheckLevel level, JSError error) {
+          throw new AssertionError(error.toString());
+        }
+      });
+      return options;
+    }
+
+    Options(String[] args) throws CmdLineException {
+      CmdLineParser parser = new CmdLineParser(this);
+      try {
+        parser.parseArgument(args);
+
+        if(arguments.isEmpty())
+          throw new CmdLineException(parser, "No files were given");
+
+      } catch (CmdLineException e) {
+        System.err.println(e.getMessage());
```

Comment:
Move this usage stuff to the main method? It's only relevant for the command line execution, so should IMHO belong with `main()`.